pkg/parser/parser_test.go

```diff
@@ -210,3 +215,84 @@ func TestParser(t *testing.T) {
 		})
 	}
 }
+
+func TestCleanYAML(t *testing.T) {
+	type args struct {
+		in []byte
+	}
+	type want struct {
+		out []byte
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"Empty": {
+			reason: "Should return nil on empty input",
+			args:   args{in: []byte("")},
+			want:   want{out: nil},
+		},
+		"CommentedOut": {
```

Comment:
good test cases! one thing that is nagging my mind here is if there's a scenario @muvaf was handling with the original function that really needed all commented lines removed.  It seems like the conclusion here is that that was too aggressive, and its OK to leave in commented lines if the full block isn't completely commented.

You feel good about that too? There is not a scenario where commented lines should actually be removed? 🤔

Reply:
From the description of the original PR my understanding was that only fully commented files were an issue, so being less aggressive should be fine. The current behavior in released versions of Crossplane is to not drop any line at all, so I guess there should be no issue only dropping fully commented files.

Reply:
@phisco is right, my motivation was to have the parser skip fully-commented documents so we don't have errors in the upstream marshal/unmarshal functions. It was essentially extending the definition of empty document from having no characters at all to having all lines commented out. Not sure why I ended up removing any lines, maybe while [trying to reduce](https://github.com/crossplane/crossplane-runtime/pull/504#discussion_r1287250192) the number of loops?

I think this function can just be a check like `isEmpty` and not make any changes to the document - the problem isn't about having comments but more about having the whole document not have any useful data hence failing the functions trying to cast it onto a struct.

Reply:
Thanks @muvaf for the feedback, I'll rename the function to IsEmpty and make sure it doesn't have any side effect 👍